### PR TITLE
Refactor classes in game.dart file to separate files

### PR DESCRIPTION
**Pull request by erf**
See #286

### CODE REVIEW

File: lib/game/simple_game.dart [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import 'base_game.dart';
+
+/// This is a helper implementation of a [BaseGame] designed to allow to easily create a game with a single component.
+///
+/// This is useful to add sprites, animations and other Flame components "directly" to your non-game Flutter widget tree, when combined with [EmbeddedGameWidget].
+class SimpleGame extends BaseGame {
+  SimpleGame(Component c) {
+    add(c);
+  }
+}
```
Comment (renancaraujo) [inline lib/game/simple_game.dart:12]: IMO this should not be on a separate file, just unnecessary indirection.

Comment (erf) [reply lib/game/simple_game.dart:12]: Should it be there at all though..?

Comment (erf) [reply lib/game/simple_game.dart:12]: Move it to `EmbeddedGameWidget`?

Comment (erf) [reply lib/game/simple_game.dart:12]: Although, it think its good to be consistent, like one class per file. Not sure if you have any style guides / rules for these kind of things?

Comment (erickzanardo) [reply lib/game/simple_game.dart:12]: I have some doubts about this class as will, I think that could be removed, as just doing the following would replace it

```dart
BaseGame()
  ..add(comp);
```

Comment (erf) [reply lib/game/simple_game.dart:12]: @erickzanardo that class is abstract now, so can't be instantiated, but i could make it non-abstract

Comment (erf) [reply lib/game/simple_game.dart]: I pushed a new commit where i remove SimpleGame and make BaseGame non-abstract. Fixed and testet with the Animation example

Comment (spydon) [reply lib/game/simple_game.dart]: Good solution!

File: lib/util.dart [hunk 2]
```diff
@@ -154,7 +155,7 @@ class Util {
   /// This is intended to be used by non-game apps that want to add a sprite sheet animation.
   widgets.Widget animationAsWidget(Position size, Animation animation) {
     return EmbeddedGameWidget(
-        SimpleGame(AnimationComponent(size.x, size.y, animation)),
+        BaseGame()..add(AnimationComponent(size.x, size.y, animation)),
         size: size);
```
Comment (luanpotter) [inline lib/util.dart:159]:
```suggestion
        size: size,
    );
```

File: doc/examples/animation_widget/lib/main.dart [hunk 3]
```diff
@@ -9,6 +9,7 @@ import 'package:flutter/material.dart';
 Sprite _sprite;
 
 void main() async {
+  WidgetsFlutterBinding.ensureInitialized();
```
Comment (luanpotter) [inline doc/examples/animation_widget/lib/main.dart:12]: do we still need this? I thought this was not necessary anymore, does this example give out an error w/o?

File: doc/examples/animation_widget/lib/main.dart [hunk 4]
```diff
@@ -71,7 +72,7 @@ class _MyHomePageState extends State<MyHomePage> {
             Flame.util.animationAsWidget(
                 _position,
                 animation.Animation.sequenced('minotaur.png', 19,
-                    textureWidth: 96.0)),
+                    textureWidth: 96.0, textureHeight: 96)),
```
Comment (luanpotter) [inline doc/examples/animation_widget/lib/main.dart:75]: let's do one param per line or all in the same line (if it fits)

File: doc/examples/animation_widget/lib/main.dart (hunk 3 above)
Comment (luanpotter) [reply doc/examples/animation_widget/lib/main.dart:12]: (if this is unrelated to your change it's fine)

File: lib/game/base_game.dart [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+/// This is a more complete and opinionated implementation of Game.
+///
+/// It still needs to be subclasses to add your game logic, but the [update], [render] and [resize] methods have default implementations.
+/// This is the recommended structure to use for most games.
+/// It is based on the Component system.
+class BaseGame extends Game with TapDetector {
+  /// The list of components to be updated and rendered by the base game.
+  OrderedSet<Component> components =
+      OrderedSet(Comparing.on((c) => c.priority()));
```
Comment (luanpotter) [inline lib/game/base_game.dart:27]: this should fit in a single line I think, no?

File: lib/game/embedded_game_widget.dart [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  RenderBox createRenderObject(BuildContext context) {
+    return RenderConstrainedBox(
+        child: GameRenderBox(context, game),
+        additionalConstraints:
+            BoxConstraints.expand(width: size?.x, height: size?.y));
+  }
+
+  @override
+  void updateRenderObject(
+      BuildContext context, RenderConstrainedBox renderBox) {
```
Comment (luanpotter) [inline lib/game/embedded_game_widget.dart:31]: single line or one param per line

File: lib/game/base_game.dart (hunk 5 above)
Comment (erf) [reply lib/game/base_game.dart:27]: i let `flutter format` decide that, don't you use it?

File: lib/game/embedded_game_widget.dart (hunk 6 above)
Comment (erf) [reply lib/game/embedded_game_widget.dart:31]: i let `flutter format` decide that, do you use default line length of 80 or?

File: doc/examples/animation_widget/lib/main.dart (hunk 4 above)
Comment (erf) [reply doc/examples/animation_widget/lib/main.dart:75]: i'll add commas

File: lib/util.dart (hunk 2 above)
Comment (erf) [reply lib/util.dart:159]: i'll add comma

File: doc/examples/animation_widget/lib/main.dart (hunk 3 above)
Comment (erf) [reply doc/examples/animation_widget/lib/main.dart:12]: I get the following error if run without `ensureInitialized()`:
```
[VERBOSE-2:ui_dart_state.cc(157)] Unhandled Exception: ServicesBinding.defaultBinaryMessenger was accessed before the binding was initialized.
If you're running an application and need to access the binary messenger before `runApp()` has been called (for example, during plugin initialization), then you need to explicitly call the `WidgetsFlutterBinding.ensureInitialized()` first.
If you're running a test, you can call the `TestWidgetsFlutterBinding.ensureInitialized()` as the first line in your test's `main()` method to initialize the binding.
#0      defaultBinaryMessenger.<anonymous closure> (package:flutter/src/services/binary_messenger.dart:76:7)
#1      defaultBinaryMessenger (package:flutter/src/services/binary_messenger.dart:89:4)
#2      PlatformAssetBundle.load (package:flutter/src/services/asset_bundle.dart:219:15)
#3      Images._fetchToMemory (package:flame/images.dart:30:46)
#4      Images.load (package:flame/images.dart:24:37)
#5      Sprite.loadSprite (package:flame/sprite.dart:47:44)
<…>
```
I'm on Flutter dev channel.

**Comment by erf** [comment]
Pushed a new commit with comma fixes

**Pull request merged**